browser/src/control/Control.Zotero.js

```diff
@@ -314,30 +316,92 @@ L.Control.Zotero = L.Control.extend({
 		};
 	},
 
-	getCitationJSONString: function(item) {
+	resetCitation: function() {
+		this.citationCluster = {};
+		this.citations = {};
+		delete this.settings.citationNumber;
+	},
+
+	setCitationNumber: function(number) {
+		if (this.settings.citationFormat !== 'numeric')
+			return;
+
+		if (!this.settings.citationNumber)
+			this.settings.citationNumber = 1;
+
+		this.settings.citationNumber = this.settings.citationNumber <= parseInt(number) ? parseInt(number) + 1 : this.settings.citationNumber;
+
+	},
+
+	getCitationText: function(citationId, text) {
+		if (this.citations[citationId])
+			return this.citations[citationId];
+
+		if (this.settings.citationFormat === 'numeric') {
+			if (!this.settings.citationNumber)
+				this.settings.citationNumber = 1;
+			this.citations[citationId] = this.settings.citationNumber++;
+		} else {
+			this.citations[citationId] = L.Util.trim(text, this.settings.layout.prefix, this.settings.layout.suffix);
+		}
+
+		return this.settings.group.prefix + this.citations[citationId] + this.settings.group.suffix;
+	},
+
+	handleCitationText: function(citationItems) {
+		//Some citations have style so trying to change just inner most text without affecting other html
+		var citationString = '';
+		var citationNode = $(new DOMParser().parseFromString(citationItems[0].citation, 'text/html').body);
+
+		var that = this;
+		citationItems.forEach(function(item) {
+			var itemHTML = $(new DOMParser().parseFromString(item.citation, 'text/html').body);
+			var innerMostNode = itemHTML.find('*:not(:has("*"))')[0];
```

Comment:
can we try to not use `jQuery` and instead use `HTMLElement.querySelector` or similar API (one dependency less)

Reply:
done